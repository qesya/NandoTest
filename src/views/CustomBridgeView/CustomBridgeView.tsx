import React from 'react';
import {StyleSheet, View, Text, StatusBar} from 'react-native';
import Button from '../../components/Button/Button';
import * as colours from '../../utilities/colours';
import * as fonts from '../../utilities/fonts';

type Props = {
  appVersion: string;
  onPressGoBack: () => void;
};

const CustomBridgeView = ({appVersion, onPressGoBack}: Props) => {
  return (
    <View style={styles.container}>
      <StatusBar
        translucent
        backgroundColor={colours.WHITE}
        barStyle="dark-content"
      />
      <View>
        <Text style={styles.txtHeader}>APP VERSION</Text>
        <Text style={styles.txtNote}>this value from native modules</Text>
        <Text style={styles.txtAppVersion}>{appVersion}</Text>
      </View>
      <Button style={styles.btn} title="Back" onPress={onPressGoBack} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 16,
    backgroundColor: colours.WHITE,
    justifyContent: 'center',
  },
  txtHeader: {
    ...fonts.PoppinsLight,
    fontSize: 24,
    marginTop: -40,
    textAlign: 'center',
    paddingHorizontal: 20,
    marginBottom: 6,
    color: colours.BLACK,
    textTransform: 'uppercase',
    letterSpacing: 2,
  },
  txtAppVersion: {
    ...fonts.PoppinsMedium,
    fontSize: 34,
    textAlign: 'center',
    paddingHorizontal: 20,
    marginTop: 18,
    color: colours.BLACK,
    textTransform: 'uppercase',
    letterSpacing: 2,
  },
  txtNote: {
    ...fonts.PoppinsLight,
    fontSize: 8,
    textAlign: 'center',
    color: colours.BLACK,
    textTransform: 'uppercase',
    letterSpacing: 2,
  },
  btn: {
    position: 'absolute',
    bottom: 30,
    left: 0,
    right: 0,
    marginHorizontal: 16,
  },
});
export default CustomBridgeView;
