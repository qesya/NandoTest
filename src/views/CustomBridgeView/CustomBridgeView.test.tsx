import {render} from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CustomBridgeView from './CustomBridgeView';

describe('CustomBridgeView', () => {
  test('renders correctly', () => {
    const component = renderer
      .create(
        <CustomBridgeView appVersion="1.0.1" onPressGoBack={() => jest.fn()} />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('app versions defined', () => {
    const appVersion = '1.0.0';
    const component = render(
      <CustomBridgeView
        appVersion={appVersion}
        onPressGoBack={() => jest.fn()}
      />,
    );
    expect(component.queryByTestId('AppVersionText')).toBe(appVersion);
  });
});
