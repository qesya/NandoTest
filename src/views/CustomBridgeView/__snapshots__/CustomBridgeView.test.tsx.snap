// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBridgeView renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#1F1F1F",
          "fontFamily": "Poppins-Light",
          "fontSize": 24,
          "letterSpacing": 2,
          "marginBottom": 6,
          "marginTop": -40,
          "paddingHorizontal": 20,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      APP VERSION
    </Text>
    <Text
      style={
        Object {
          "color": "#1F1F1F",
          "fontFamily": "Poppins-Light",
          "fontSize": 8,
          "letterSpacing": 2,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      this value from native modules
    </Text>
    <Text
      style={
        Object {
          "color": "#1F1F1F",
          "fontFamily": "Poppins-Medium",
          "fontSize": 34,
          "letterSpacing": 2,
          "marginTop": 18,
          "paddingHorizontal": 20,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#1F1F1F",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 10,
        },
        Object {
          "bottom": 30,
          "left": 0,
          "marginHorizontal": 16,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
        }
      }
    >
      Back
    </Text>
  </View>
</View>
`;
